netplan/cli/commands/try_command.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/python3
+#
+# Copyright (C) 2018 Canonical, Ltd.
+# Author: Mathieu Trudel-Lapierre <mathieu.trudel-lapierre@canonical.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; version 3.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+'''netplan try command line'''
+
+import fcntl
+import os
+import termios
+import time
+import select
+import signal
+import sys
+
+from netplan.configmanager import ConfigManager
+import netplan.cli.utils as utils
+from netplan.cli.commands.apply import NetplanApply
+
+# Keep a timeout long enough to allow the network to converge, 60 seconds may
+# be slightly short given some complex configs, i.e. if STP must reconverge.
+INPUT_TIMEOUT = 120
+
+
+class NetplanTry(utils.NetplanCommand):
+
+    def __init__(self):
+        super().__init__(command_id='try',
+                         description='Try to apply a new netplan config to running '
+                                     'system, with automatic rollback',
+                         leaf=True)
+        self.configuration_changed = False
+        self.config_manager = ConfigManager()
+
+    def run(self):  # pragma: nocover (requires user input)
+        self.parser.add_argument('--config-file',
+                                 help='Apply the config file in argument on top of current configuration.')
+
+        self.func = self.command_try
+
+        self.parse_args()
+        self.run_command()
+
+    def command_try(self):  # pragma: nocover (requires user input)
+        try:
+            fd = sys.stdin.fileno()
```

Comment:
could you put the terminal mangling in another method to make the overall structure here clearer to read?